src/pocketmine/entity/MobsControl.php

```diff
@@ -0,0 +1,368 @@
+<?php
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+namespace pocketmine\entity;
+
+use pocketmine\entity\Monster;
+use pocketmine\utils\MainLogger;
+
+class MobsControl {
+	
+	const STATE_KILL = 1;
+	const STATE_SLEEP = 2;
+	const STATE_ACTIVE = 3;
+	
+	const FOLLOW_NORMAL = 1;
+	const FOLLOW_TARGET = 2;
+	
+	private $types = array("Zombie", "Creeper", "Skeleton", "Spider", "ZombiePigman", "Slime", "Enderman", "Silverfish",
+			"CaveSpider", "Ghast", "MagmaCube", "Blaze", "ZombieVillager");
+	
+	private static $instance;
+	
+	private $state = MobsControl::STATE_SLEEP; 
+	private $speedMap;
+	private $damageMap;
+	private $healthMap;
+	private $proximityMap;
+	private $followMap;
+	
+	private $countMap;
+	
+	public static function getInstance() {
+		if (null === MobsControl::$instance) {
+			MobsControl::$instance = new MobsControl();
+		}
+		return MobsControl::$instance;
+	}
+	
+	protected function __construct() {
+		$this->speedMap["Default"] = .1;
+		$this->damageMap["Default"] = 1;
+		$this->healthMap["Default"] = 20;
+		$this->proximityMap["Default"] = 10;
+		$this->followMap["Default"] = MobsControl::FOLLOW_NORMAL;
+		$this->countMap["PlaceHolder"] = 0;
+		
+		
+// 		$this->speedMap["Zombie"] = .1;
```

Comment:
You should actually put these in their classes, just like how gravity and drag is overridden in those classes.